apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -44,6 +66,63 @@ export const askAITutor = createAsyncThunk(
   }
 );
 
+// General Chat
+// This thunk's callback function submits a user chat message to the chat completion endpoint,
+// waits for a chat completion response, and updates the user message state.
+export const submitChatMessage = createAsyncThunk(
```

Comment:
I kept this separate because it handles all of the work for a series of back-and-forth messages rather than a single ask and response for compilation and validation. Conceptually we could probably just have one thunk that handles all 3 cases if folks have strong opinions about it.